packages/visual-editor/vite.config.ts

```diff
@@ -5,14 +5,27 @@ import type { Plugin } from "vite";
 import { exec } from "node:child_process";
 
 export default defineConfig(() => ({
+  define: {
+    'process.env.PLUGIN': 'false',
+  },
   plugins: [react(), dts()],
   build: {
     lib: {
-      entry: path.resolve(__dirname, "src/index.ts"),
+      entry: {
+        "visual-editor": path.resolve(__dirname, "src/index.ts"),
+        "plugin/plugin": path.resolve(__dirname, "src/vite-plugin/index.ts"),
```

Comment:
You have a separate vite config for the plugin. This one shouldn't also be building the plugin.